NineChronicles.DataProvider/DataRendering/StakeData.cs

```diff
@@ -22,18 +18,36 @@ public static class StakeData
             DateTimeOffset blockTime
         )
         {
-            outputStates.TryGetStakeState(signer, out StakeState stakeState);
-            var prevStakeStartBlockIndex =
-                !previousStates.TryGetStakeState(signer, out StakeState prevStakeState)
-                    ? 0 : prevStakeState.StartedBlockIndex;
-            var newStakeStartBlockIndex = stakeState.StartedBlockIndex;
             var currency = outputStates.GetGoldCurrency();
-            var balance = outputStates.GetBalance(signer, currency);
-            var stakeStateAddress = StakeState.DeriveAddress(signer);
-            var previousAmount = previousStates.GetBalance(stakeStateAddress, currency);
-            var newAmount = outputStates.GetBalance(stakeStateAddress, currency);
+            var stakeAddress = StakeStateV2.DeriveAddress(signer);
+            FungibleAssetValue previousAmount;
+            FungibleAssetValue newAmount;
+            long prevStakeStartBlockIndex;
+            long newStakeStartBlockIndex;
+            if (previousStates.TryGetStakeStateV2(signer, out var prevStakeStateV2))
+            {
+                previousAmount = previousStates.GetBalance(stakeAddress, currency);
+                prevStakeStartBlockIndex = prevStakeStateV2.StartedBlockIndex;
+            }
+            else
+            {
+                previousAmount = new FungibleAssetValue(currency, 0, 0);
+                prevStakeStartBlockIndex = 0;
+            }
 
-            var stakeModel = new StakeModel()
+            if (!outputStates.TryGetStakeStateV2(signer, out var stakeStateV2))
```

Comment:
Should this part just be `if (outputStates.TryGetStakeStateV2(signer, out var stakeStateV2))` without the `!`, so that we can get the `newAmount` and `newStakeStartBlockIndex` if the stateStake exists?

Reply:
Oh, thanks!